tests/cypress/e2e/tools/classes/class.js

```diff
@@ -3,22 +3,40 @@ import { goToTeachersPage } from "../navigation/nav";
 export function createClass()
 {
     const classname = `test class ${Math.random()}`;
-    
+
     goToTeachersPage();
     cy.wait(500);
 
     cy.get('#create_class_button').click();
     cy.get('#modal-prompt-input').type(classname);
     cy.get('#modal-ok-button').click();
-    
+
     goToTeachersPage();
     cy.wait(500);
 
     return classname;
 }
 
+/**
+ * Make sure that at least one class exists
+ *
+ * Create a class if one doesn't exist already.
+ */
+export function ensureClass()
+{
+    const classname = `test class ${Math.random()}`;
+    goToTeachersPage();
+
+    const viewClassLink = cy.getBySel('view_class_link');
```

Comment:
Hello @rix0rrr this is something I missed yesterday, and just realized now trying to incorporate this on another refactor in #4032 . You can't assign the return value of a Cypress command to variable because they are ran asynchronously. There's more info here: https://docs.cypress.io/guides/core-concepts/variables-and-aliases

Reply:
I propose we change it to this, since we only need the value in the function `createClassAndAddStudents`, which is used in create_accounts.cy,js and I think we can just put that code in the `beforeEach` of that particular file as suggested by the cypress documentation:

```
export function ensureClass()
{
    goToTeachersPage();
    if (cy.getBySel('view_class_link').length === 0) {
        createClass();
    }
}
```

Reply:
Good call. For now, I turned this into a function that returns a `Promise`, that can be `await`ed in the caller:

```ts
function ensureClass() {
  return new Promise(ok => {
    cy.get('#whatever').then(x => ok(x));
  });
}

beforeEach(async () => {
   await ensureClass();
});
```

Should be good enough for now, while you do the refactor?

Reply:
Yes this is good enough for now!